### DISCUSSION TITLE
How to securely get a user their collection

### DISCUSSION THREAD

**Discussion by HexiDev** [pocketbase/pocketbase#2632] [Q&A] [answered]
So I am using nextjs and am trying to know serverside if the user is valid or not with a fetch request

A friend did this:
"You do server-side 'loadfromcookie,' and then it has a session. Then server-side 'pb.collection' and so on to retrieve the user from the database. And then check if the data is correct."

But I dont think thats the best way for it but I am using SSR with cookies in nextjs so thats why he did that

**Comment by ganigeorgiev** [comment, marked as answer]
> and am trying to know serverside if the user is valid or not with a fetch request

After you have loaded the cookie in `pb.authStore` you can call `authRefresh()` to validate the token and fetch the latest user data (it will automatically update the `pb.authStore`):

```js
pb.authStore.loadFromCookie(...);


try {
    // get an up-to-date auth store state by verifying and refreshing the loaded auth model (if any)
    pb.authStore.isValid && await pb.collection('users').authRefresh();
} catch (_) {
    // clear the auth store on failed refresh
    pb.authStore.clear();
}

...
```

If you haven't already, you can also check the SSR integration examples in https://github.com/pocketbase/js-sdk#ssr-integration (_we don't have yet examples for Next.js 13 and the new app directory but you can use them as a reference_).

**Reply by HexiDev** [reply]
Thanks for the fast response I did see authRefresh but didnt think a lot of it... But this helped a lot thanks!
